src/main/java/net/atos/zac/flowable/TakenService.java

```diff
@@ -168,28 +167,26 @@ public TaskInfo readTask(final String taskId) {
             return task;
         }
         final HistoricTaskInstance historicTaskInstance = findClosedTask(taskId);
-        if (historicTaskInstance != null) {
-            return historicTaskInstance;
+        if (historicTaskInstance == null) {
+            throw new RuntimeException(String.format("No task found with task id '%s'", taskId));
         }
-        throw new RuntimeException(String.format("No task found with task id '%s'", taskId));
+        return historicTaskInstance;
     }
 
     public Task readOpenTask(final String taskId) {
         final Task task = findOpenTask(taskId);
-        if (task != null) {
-            return task;
-        } else {
+        if (task == null) {
             throw new RuntimeException(String.format("No open task found with task id '%s'", taskId));
         }
+        return task;
     }
 
     public HistoricTaskInstance readClosedTask(final String taskId) {
         final HistoricTaskInstance historicTaskInstance = findClosedTask(taskId);
-        if (historicTaskInstance != null) {
-            return historicTaskInstance;
-        } else {
+        if (historicTaskInstance == null) {
             throw new RuntimeException(String.format("No closed task found with task id '%s'", taskId));
```

Comment:
I don't know who calls this method but the caller might not expect this exception when the task is not really closed? They might now expect a null? Good candidate for some JavaDoc maybe?

Reply:
https://dimpact.atlassian.net/browse/PZ-1778